### DISCUSSION TITLE
GeoJSON to GeoShapeQuery

### DISCUSSION THREAD

**Discussion by cramey** [blevesearch/bleve#1926] [Q&A] [answered]
Reading over the API documentation, official docs, and the issue list, it is unclear to me how to generate a GeoShapeQuery from GeoJSON. You can very nearly directly insert GeoJSON into an index, but it appears to want a completely different format for a query. Can someone provide an example of turning GeoJSON into a valid GeoShapeQuery?

**Comment by metonymic-smokey** [comment]
@cramey apologies for the delayed response.
From what I understand, your ask is to convert GeoJSON(provided as a string) into a query.
I think this unit test does just that: https://github.com/blevesearch/bleve/blob/master/search/query/query_test.go#L454-L765. Pls take a look at one of the cases and let me know if you need any more info.

thanks!

**Reply by cramey** [reply, marked as answer]
What that testing function does is unmarshal several GeoShapeQuerys to compare expected results with output. It's not a complete answer because it uses query.ParseQuery() and expects JSON compatible with a GeoShapeQuery, which includes GeoJSON, but isn't exclusively GeoJSON. It did lead me around to geo.ParseGeoJSONShape(), which can be used to Unmarshal GeoJSON into struct compatible with a GeoShapeQuery. Here's my code:

```
shape, err := geo.ParseGeoJSONShape([]byte(geojson))
if err != nil {
	// Handle error
}

gs_query := &query.GeoShapeQuery{
	FieldVal: "geog",
	Geometry: query.Geometry{
		Shape: shape,
		Relation: "intersects",
	},
}
```

**Comment by metonymic-smokey** [comment]
I picked that test since it pointed to a function which unmarshaled strings with GeoJSON into valid geoshape queries.

Great to see that found the answer, closing this as resolved.
